src/main/resources/archetype-resources/frontend/src/main/kotlin/component/pages/mainPage/MainPage.kt

```diff
@@ -0,0 +1,38 @@
+package component.pages.mainPage
+
+import app.AppState
+import react.*
+import react.dom.*
+import react.redux.rConnect
+import redux.*
+import imports.ajs.getCurrentUsername
+
+interface MainPageStateProps : RProps {
+    var chaosMode: Boolean
+}
+
+interface MainPageDispatchProps : RProps
+interface MainPageProps : MainPageStateProps, MainPageDispatchProps
+
+val mainPage: RClass<MainPageProps> =
+    rConnect<AppState, RAction, WrapperAction, RProps, MainPageStateProps, MainPageDispatchProps, MainPageProps>(
+        { state, _ ->
+            chaosMode = state.chaosMode
+        },
+        { _, _ -> }
+    )(MainPage::class.js.unsafeCast<RClass<MainPageProps>>())
+
+class MainPage(props: MainPageProps) : RComponent<MainPageProps, MainPage.State>(props) {
+
+
+    override fun RBuilder.render() {
+        div {
```

Comment:
Shouldn't this be something like this:

div {
            attrs.id = "content"
            props.screen.render(this)
        }
        renderNotifications()
        
and the default state for Screen is Main

Reply:
The Applikation.main() attaches the app as a whole to the DOM element named content. And this special element is created outside the frontend, so that you have a choice where to include it. It is part of module.vm:
```<section id="content" role="main">```

The div here must not have the same id, as having the same id multiple times in the DOM is not allowed.

But maybe we should not call it content, as that is a quite generic name that does not tell what is going on here. Maybe rather something like frontend-app-main or something like that.

Also interesting will be how to start different parts of the application. E.g. like you have a config area and a room book area. This should be easier than having to replicate the whole app and all its infrastructure. We could have separate entry points. But this would be something I'd rather tackle in a subsequent pull request.